contracts/chain-adapters/DoctorWho_Adapter.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./interfaces/AdapterInterface.sol";
+import "../external/interfaces/WETH9Interface.sol";
+
+// @dev Use local modified CrossDomainEnabled contract instead of one exported by eth-optimism because we need
+// this contract's state variables to be `immutable` because of the delegateCall call.
+import "./CrossDomainEnabled.sol";
+import "@eth-optimism/contracts/L1/messaging/IL1StandardBridge.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../libraries/CircleCCTPAdapter.sol";
+import "../external/interfaces/CCTPInterfaces.sol";
+
+/**
+ * @notice Contract containing logic to send messages from L1 to Base. This is a modified version of the Optimism adapter
```

Comment:
@dohaki There are a few residual references to `Base` here. They seem to be only in the comments so they're not critical; I just wanted to flag them.